### ISSUE TITLE
Intermittent UI test failure - org.mozilla.samples.glean.pings.BaselinePingTest  validateBaselinePing

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla-mobile/android-components#9714]
1 failure so far.

#9705 and #9707 had landed yesterday. cc @csadilek 

### Firebase Test Run:
https://console.firebase.google.com/project/moz-android-components-230120/testlab/histories/bh.acacddb44ee28c64/matrices/5612267530912440552/executions/bs.a89322c7ddbff8d2
### Stacktrace:
```
java.lang.NullPointerException: request must not be null
at org.mozilla.samples.glean.pings.BaselinePingTest.waitForPingContent(BaselinePingTest.kt:93)
at org.mozilla.samples.glean.pings.BaselinePingTest.waitForPingContent$default(BaselinePingTest.kt:86)
at org.mozilla.samples.glean.pings.BaselinePingTest.validateBaselinePing(BaselinePingTest.kt:126)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:154)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1837)
```
### Build: 
Nexus 6, Virtual, API Level 21

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-8244)

**Comment by csadilek** [comment]
@mdboom @badboy Could you take a look at this? I fixed the test here in A-C following the Fenix patch: https://github.com/mozilla-mobile/android-components/pull/9707

**Comment by badboy** [comment]
Checking.
That stacktrace looks slightly different, so I'm inclined to think there's another problem.

**Comment by badboy** [comment]
I wasn't able to reproduce that locally. I ran the tests several dozen times now (`./gradlew -Pandroid.testInstrumentationRunnerArguments.class=org.mozilla.samples.glean.pings.BaselinePingTest samples-glean:connectedDebugAndroidTest`), but no failure.
I also did some manual testing on the sample app (enabled logging with `adb shell am start -n org.mozilla.samples.glean.debug/mozilla.telemetry.glean.debug.GleanDebugActivity --ez logPings true`, then reopened the app, put it to background etc.)

That all looks good.
The only way the above can happen is if there's no baseline produced within the 20s timeout.

Maybe we got unlucky and something else went wrong.
How often do these tests run? Can we keep an eye on them and see if it fails again before I spend more time on this?
